File: 0695-max-area-of-island/0695-max-area-of-island.cpp
class Solution {
public:
    int maxAreaOfIsland(vector<vector<int>>& grid) {
        int ans = 0;
        int row = grid.size(), col = grid[0].size();
        int offset[] = {1,0,-1,0,1};
        for(int i = 0; i<row; i++){
            for(int j = 0; j<col; j++){
                if(grid[i][j] == 1){
                    queue<pair<int,int>> q;
                    q.push({i,j});
                    grid[i][j] = 0;
                    int cnt = 0;
                    while(!q.empty()){
                        cnt++;
                        int curr_row = q.front().first;
                        int curr_col = q.front().second;
                        q.pop();
                        for(int k = 0; k<4; k++){
                            int r = curr_row + offset[k];
                            int c = curr_col + offset[k+1];
                            if(r < 0 || c < 0 || r >= row || c >= col || grid[r][c] == 0) continue;
                            grid[r][c]  = 0;
                            q.push({r,c});
                        }
                    }
                    ans = fmax(ans,cnt);
                }
            }
        }
    return ans;
    }
};

File: 0695-max-area-of-island/0695-max-area-of-island.js
/**
 * @param {number[][]} grid
 * @return {number}
 */
var maxAreaOfIsland = function(grid) {
    const [row,col] = [grid.length, grid[0].length];
    let [ans,cnt] = [0,0];
    const dfs = function(r,c){
        if(r < 0 || c < 0 || r>= row || c >= col || grid[r][c] == 0) return;
        grid[r][c] = 0;
        cnt++;
        dfs(r+1,c);
        dfs(r-1,c);
        dfs(r,c-1);
        dfs(r,c+1);
    }
    for(let i = 0; i<row; i++){
        for(let j = 0; j<col; j++){
            if(grid[i][j] == 1){
                cnt = 0;
                dfs(i,j)
                ans = Math.max(ans,cnt);
            }
        }
    }
    return ans;
};